### PR TITLE
sinon - defined SinonSpy.wrappedMethod

### PR DESCRIPTION
**Pull request by NoamDev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.

- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by NoamDev** [comment]
Based on sinonjs/sinon#988.
Note that this feature is not yet documented but I made an issue for that here: sinonjs/sinon#2182

File: types/sinon/ts3.1/index.d.ts [hunk 1]
```diff
@@ -334,7 +334,7 @@ declare namespace Sinon {
          */
         restore(): void;
         /**
-         * Get the original method. Only available if the spy replaced an existing method.
+         * Get the original method. Only available if the spy replaced an existing method, but not a property.
```
Comment (43081j) [inline types/sinon/ts3.1/index.d.ts]: are you sure this is true? if you spy on an accessor (i.e a property), the wrapped accessor would have a wrappedMethod wouldn't it?

as in:

```
const spy = sinon.spy(obj, 'prop', ['get']);
spy.get.wrappedMethod;
```

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
If it's not documented in a project with otherwise comprehensive documentation, it's likely not intended to be used. Please reopen when they actually document the method for public use.

**Comment by NoamDev** [comment]
I'm closing it by myself 'cause  I'm tired from all the notifications.
Anyway. Let's wait to see whther or not sinon will document that feature

**Comment by mroderick** [comment]
The PR for Sinon has been merged, it is not yet documented on the website (because updates there are tied to releases ... for legacy reasons that I'll be addressing soon), but it is visible in `master`.

Is that enough to unblock the work here?

**Comment by 43081j** [comment]
bleh

@RyanCavanaugh  mind taking a look? bot will keep closing it i imagine

**Comment by mroderick** [comment]
That bot is keen! Perhaps I need to install that on a few repositories ...

**Pull request closed without merging**